BootloaderCommonPkg/Include/Library/SecureBootLib.h

```diff
@@ -61,6 +58,9 @@ DoHashVerify (
   @param[in]  Signature       Signature for the data buffer.
   @param[in]  PubKey          Public key data pointer.
   @param[in]  PubKeyHash      Public key hash value when ComponentType is not used.
+  @param[in]  PubKeyHashAlg   Pubkey Hash Alg .
```

Comment:
This method is doing too many things and there are too many arguments.
Should we divide this method into multiple simpler methods?
DoRsaVerifyComponent()
DoRsaVerifyData()

Are there any use-cases where current api is being used in non-component scenarios?

Reply:
Instead of splitting this API, could we define a new param struct for combining args together?

Reply:
I still feel we need separate API here to differentiate Comp and non-comp use case for a simpler API interface to consumer.

Reply:
Will separate API's for 
DoVerifyComponent() ==> Hash Verification
DoRsaVerifyData() ==> Rsa Verification

Reply:
We have DoHashVerify() already.  What is the difference between DoHashVerify/DoVerifyComponent ?

Reply:
Discussed with Sachin, I will come up design for adding data structures for Signature & Public key and updated API interfaces.